services/management/actions.go

```diff
@@ -0,0 +1,217 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package management
+
+import (
+	"context"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/percona/pmm/api/managementpb"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+var (
+	errUnsupportedAction  = errors.New("unsupported action")
+	errPmmAgentIDNotFound = errors.New("can't detect pmm_agent_id")
+)
+
+type agentsRegistry interface {
+	// SendRequest sends request to pmm-agent with given id.
+	SendRequest(ctx context.Context, pmmAgentID string, payload agentpb.ServerRequestPayload) agentpb.AgentResponsePayload
+}
+
+// ActionsService describes an Actions Application Service.
+// Provides functions for PMM Actions manipulation.
+type ActionsService struct {
+	agentsRegistry agentsRegistry
+	actionsStorage *InMemoryActionsStorage
+	logger         logrus.FieldLogger
+	db             *reform.DB
+}
+
+// NewActionsService creates new actions service.
+func NewActionsService(r agentsRegistry, s *InMemoryActionsStorage, db *reform.DB) *ActionsService {
+	return &ActionsService{
+		agentsRegistry: r,
+		actionsStorage: s,
+		logger:         logrus.WithField("component", "actions-service"),
+		db:             db,
+	}
+}
+
+// RunActionParams parameters for run actions.
+type RunActionParams struct {
+	ActionName   managementpb.ActionType
+	ActionParams []string
+	PmmAgentID   string
+	NodeID       string
+	ServiceID    string
+}
+
+// RunAction runs PMM Action on the given client.
+func (a *ActionsService) RunAction(ctx context.Context, rp RunActionParams) (actionID string, errorVar error) {
```

Comment:
Accepter structure by pointer – it already contains non-value type (slice).
Please avoid named result parameters if they are not used.

Reply:
> Accepter structure by pointer – it already contains non-value type (slice).
What do you mean? Should I pass `RunActionParams` by pointer?

> Please avoid named result parameters if they are not used.
I'am avoid. Here I'm using it for clean API (self-documented). To display, I return not `string` by `ActionID`.
<img width="544" alt="clear_api" src="https://user-images.githubusercontent.com/1532188/58022156-7fca3a00-7b15-11e9-8321-596ede5f98ea.png">